activation/atxdb.go

```diff
@@ -446,6 +472,13 @@ func (db *ActivationDb) storeAtxUnlocked(atx *types.ActivationTx) error {
 	if err != nil {
 		return err
 	}
+
+	// notify subscribers
+	if ch, found := db.atxChannels[atx.Id()]; found {
+		close(ch)
+		delete(db.atxChannels, atx.Id())
```

Comment:
you delete from map with no lock

Reply:
This is inside the method `storeAtxUnlocked` which is only called under lock

Reply:
is it the same lock under which atxchannels are added?